padlockcloud/server_test.go

```diff
@@ -503,6 +510,39 @@ func TestAuthentication(t *testing.T) {
 		res, _ = ctx.request("GET", ctx.host+"/activate/?t=asdf", "", ApiVersion)
 		testError(t, res, &BadRequest{"invalid activation token"})
 	})
+
+	t.Run("whitelist", func(t *testing.T) {
+		ctx.resetAll()
+
+		dir, err := ioutil.TempDir("", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer os.RemoveAll(dir)
+
+		tmpFile := filepath.Join(dir, "tmpFile")
+		d1 := []byte(testEmail + "\n")
+		if err := ioutil.WriteFile(tmpFile, d1, 0644); err != nil {
+			t.Fatalf("Error writing to whitelist file: %s", err.Error())
+		}
+
+		//setup config with whitelist
+		ctx := newServerTestContextWithConfig(&ServerConfig{WhitelistPath: tmpFile})
+		ctx.followRedirects(false)
+
+		randomEmail := "random@random.com"
+		if _, err = ctx.loginApi(randomEmail); err == nil {
+			t.Errorf("Should have failed to login because %s is not on whitelist", randomEmail)
+		}
+
+		if _, err = ctx.loginApi(testEmail); err != nil {
+			t.Errorf("Should have been able to login because %s is on whitelist: %s", whitelistTestEmail, err.Error())
+		}
+
+		//setup old server for future tests
+		ctx = newServerTestContext()
+		ctx.followRedirects(false)
```

Comment:
Don't need this, since `ctx` is scoped to this block, thus you're not leaking state (see previous comment). Again, a different variable name could be helpful here to avoid confusion.